tests/add_fee_amount.go

```diff
@@ -0,0 +1,115 @@
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	teleportermessenger "github.com/ava-labs/teleporter/abi-bindings/go/Teleporter/TeleporterMessenger"
+	"github.com/ava-labs/teleporter/tests/network"
+	"github.com/ava-labs/teleporter/tests/utils"
+	localUtils "github.com/ava-labs/teleporter/tests/utils/local-network-utils"
+	"github.com/ethereum/go-ethereum/common"
+	. "github.com/onsi/gomega"
+)
+
+func AddFeeAmountGinkgo() {
+	AddFeeAmount(&network.LocalNetwork{})
+}
+
+func AddFeeAmount(network network.Network) {
+	subnets := network.GetSubnetsInfo()
+	subnetAInfo := subnets[0]
+	subnetBInfo := subnets[1]
+	teleporterContractAddress := network.GetTeleporterContractAddress()
+	fundedAddress, fundedKey := network.GetFundedAccountInfo()
+	ctx := context.Background()
+
+	subnetATeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress,
+		subnetAInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+	subnetBTeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress,
+		subnetBInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+
+	// Use mock token as the fee token
+	mockTokenAddress, mockToken := localUtils.DeployExampleERC20(
+		context.Background(),
+		fundedAddress,
+		fundedKey,
+		subnetAInfo,
+	)
+	localUtils.ExampleERC20Approve(
+		ctx,
+		mockToken,
+		teleporterContractAddress,
+		big.NewInt(0).Mul(big.NewInt(1e18),
+			big.NewInt(10)),
+		subnetAInfo,
+		fundedAddress,
+		fundedKey,
+	)
+
+	initFeeAmount := big.NewInt(1)
+	// Send a transaction to Subnet A to issue a Warp Message from the Teleporter contract to Subnet B
+	sendCrossChainMessageInput := teleportermessenger.TeleporterMessageInput{
+		DestinationChainID: subnetBInfo.BlockchainID,
+		DestinationAddress: fundedAddress,
+		FeeInfo: teleportermessenger.TeleporterFeeInfo{
+			ContractAddress: mockTokenAddress,
+			Amount:          initFeeAmount,
+		},
+		RequiredGasLimit:        big.NewInt(1),
+		AllowedRelayerAddresses: []common.Address{},
+		Message:                 []byte{1, 2, 3, 4},
+	}
+
+	sendCrossChainMsgReceipt, messageID := utils.SendCrossChainMessageAndWaitForAcceptance(
+		ctx, subnetAInfo, subnetBInfo, sendCrossChainMessageInput, fundedAddress, fundedKey, subnetATeleporterMessenger)
+
+	// Add fee amount
+	additionalFeeAmount := big.NewInt(2)
+	utils.SendAddFeeAmountAndWaitForAcceptance(
+		ctx,
+		subnetAInfo,
+		subnetBInfo,
+		messageID,
+		additionalFeeAmount,
+		mockTokenAddress,
+		fundedAddress,
+		fundedKey,
+		subnetATeleporterMessenger,
+	)
+
+	// Relay message from SubnetA to SubnetB
+	network.RelayMessage(ctx, sendCrossChainMsgReceipt, subnetAInfo, subnetBInfo, false, true)
+
+	// Check Teleporter message received on the destination
+	delivered, err := subnetBTeleporterMessenger.MessageReceived(&bind.CallOpts{}, subnetAInfo.BlockchainID, messageID)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	// Send message from SubnetB to SubnetA
```

Comment:
can you elaborate that this message back will include the Teleporter receipt for previous delivery, etc.